prov/sockets/src/sock_progress.c

```diff
@@ -2150,7 +2150,7 @@ static int sock_pe_progress_rx_pe_entry(struct sock_pe *pe,
 
 	if (sock_comm_is_disconnected(pe_entry)) {
 		SOCK_LOG_DBG("conn disconnected: removing fd from pollset\n");
-		if (pe_entry->conn->sock_fd != -1) {
+		if (pe_entry->ep_attr->cmap.used > 0) {
```

Comment:
Can the cmap entry be re-used before the code gets here, possibly resulting in the entry being removed erroneously?